src/tracing/scopes.h

```diff
@@ -0,0 +1,116 @@
+/*
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+namespace chip {
+namespace Tracing {
+
+/// Trace scopes defined as an enumeration, since various tracing back-ends
+/// may need to use constant strings for tracing.
+///
+/// As a result, tracing scopes in CHIP are from a known list.
+enum class Scope
+{
+    CASESession_EstablishSession,
+    CASESession_HandleSigma1,
+    CASESession_HandleSigma1_and_SendSigma2,
+    CASESession_HandleSigma2,
+    CASESession_HandleSigma2_and_SendSigma3,
+    CASESession_HandleSigma2Resume,
+    CASESession_HandleSigma3,
```

Comment:
Please add an ` = X` to each and start at 1, with 0 being undefined. Will make it easier to look at trace values in logs and correlate.